File: app/src/main/java/deakin/gopher/guardian/view/general/CustomHeader.java
package deakin.gopher.guardian.view.general;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;
import deakin.gopher.guardian.R;

public class CustomHeader extends FrameLayout {
  public ImageView menuButton;
  private ImageView headerLogo;
  private ImageView headerProfileIcon;
  private ImageView headerTopImage;
  private TextView headerTextView;
  private View headerCard;

  public CustomHeader(Context context) {
    super(context);
    init(context, null);
  }

  public CustomHeader(Context context, AttributeSet attrs) {
    super(context, attrs);
    init(context, attrs);
  }

  public CustomHeader(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
    init(context, attrs);
  }

  private void init(Context context, AttributeSet attrs) {
    LayoutInflater.from(context).inflate(R.layout.custom_header, this, true);

    headerLogo = findViewById(R.id.headerGuardiansLogo);
    menuButton = findViewById(R.id.headerMenuIcon);
    headerProfileIcon = findViewById(R.id.headerProfileIcon);
    headerTextView = findViewById(R.id.headerTextView);
    headerCard = findViewById(R.id.headerCard);
    headerTopImage = findViewById(R.id.headerTopImage);
  }

  public void setHeaderTopImage(int resource) {
    headerTopImage.setImageResource(resource);
  }

  public void setHeaderTopImageVisibility(int visible) {
    headerTopImage.setVisibility(visible);
  }

  public void setHeaderText(String text) {
    headerTextView.setText(text);
  }

  public void setHeaderHeight(int height) {
    headerCard.getLayoutParams().height = height;
    headerCard.requestLayout();
  }
}

File: app/src/main/java/deakin/gopher/guardian/view/general/CustomHeader.kt
package deakin.gopher.guardian.view.general

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import deakin.gopher.guardian.R

class CustomHeader : FrameLayout {
    var menuButton: ImageView? = null
    private var headerLogo: ImageView? = null
    private var headerProfileIcon: ImageView? = null
    private var headerTopImage: ImageView? = null
    private var headerTextView: TextView? = null
    private var headerCard: View? = null

    constructor(context: Context) : super(context) {
        init(context, null)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        LayoutInflater.from(context).inflate(R.layout.custom_header, this, true)
        headerLogo = findViewById(R.id.headerGuardiansLogo)
        menuButton = findViewById(R.id.headerMenuIcon)
        headerProfileIcon = findViewById(R.id.headerProfileIcon)
        headerTextView = findViewById(R.id.headerTextView)
        headerCard = findViewById(R.id.headerCard)
        headerTopImage = findViewById(R.id.headerTopImage)
    }

    fun setHeaderTopImage(resource: Int) {
        headerTopImage!!.setImageResource(resource)
    }

    fun setHeaderTopImageVisibility(visible: Int) {
        headerTopImage!!.visibility = visible
    }

    fun setHeaderText(text: String?) {
        headerTextView!!.text = text
    }

    fun setHeaderHeight(height: Int) {
        headerCard!!.layoutParams.height = height
        headerCard!!.requestLayout()
    }
}